src/prediction_service_utils.cpp

```diff
@@ -82,7 +87,12 @@ std::optional<Dimension> getRequestBatchSize(const tensorflow::serving::PredictR
         SPDLOG_DEBUG("Failed to get batch size of a request. Batch size index out of shape range. Validation of request failed");
         return std::nullopt;
     }
-    return Dimension(requestInput.tensor_shape().dim(batchSizeIndex).size());
+    auto value = requestInput.tensor_shape().dim(batchSizeIndex).size();
```

Comment:
needs tests for those utils

Reply:
removed value check since the contract is to use this function only after passed validation